src/bp/nlu-core/intents/svm-intent-classifier.ts

```diff
@@ -0,0 +1,130 @@
+import { MLToolkit } from 'botpress/sdk'
+import _ from 'lodash'
+import { ListEntityModel, PatternEntity, Tools } from 'nlu-core/typings'
+import Utterance from 'nlu-core/utterance/utterance'
+
+import { IntentClassifier, IntentPredictions, IntentTrainInput } from './intent-classifier'
+
+interface Model {
+  svmModel: string | undefined
+  intentNames: string[]
+  list_entities: ListEntityModel[]
+  pattern_entities: PatternEntity[]
+}
+
+interface Predictors {
+  svm: MLToolkit.SVM.Predictor | undefined
+  intentNames: string[]
+  list_entities: ListEntityModel[]
+  pattern_entities: PatternEntity[]
+}
+
+type Featurizer = (u: Utterance, entities: string[]) => number[]
+
+export class SvmIntentClassifier implements IntentClassifier {
+  private model: Model | undefined
+  private predictors: Predictors | undefined
+
+  constructor(private tools: Tools, private featurizer: Featurizer) {}
+
+  async train(input: IntentTrainInput, progress: (p: number) => void): Promise<void> {
+    const { intents, nluSeed, list_entities, pattern_entities } = input
+
+    const entitiesName = this._getEntitiesName(list_entities, pattern_entities)
+
+    const points = _(intents)
+      .flatMap(({ utterances, name }) => {
+        return utterances.map(utt => ({
+          label: name,
+          coordinates: this.featurizer(utt, entitiesName)
+        }))
+      })
+      .filter(x => x.coordinates.filter(isNaN).length === 0)
+      .value()
+
+    const classCount = _.uniq(points.map(p => p.label)).length
```

Comment:
I know, there's the `_.uniqBy` here. I havent had time to adress previous comments yet.